README.md

```diff
@@ -1,8 +1,362 @@
+  return 0;
+}
+```
+
+## Getting Started
+
+### Users
+
+Generally speaking, because CCCL is a header-only library, users need only concern themselves with how they get the header files and how they incorporate them into their build system. 
+Anyone interested in using CCCL in their CUDA C++ application can get started by referring to the information below. 
+
+#### CUDA Toolkit 
+The easiest way to get started using CCCL is if you already have the [CUDA Toolkit](https://developer.nvidia.com/cuda-toolkit), then you already have the CCCL headers installed on your system.
+When you compile with `nvcc` you can simply include the desired headers in your code with no additional configuration required.
+If compiling with another compiler, you will need to add the include path (e.g., `/usr/local/cuda/include`) to the CCCL headers to your build system.
+
+```cpp
+#include <thrust/device_vector.h>
+#include <cub/cub.cuh>
+#include <cuda/std/atomic>
+```
+
+#### GitHub
+
+For users that want to stay on the cutting edge of CCCL development, we actively support and encourage users to use CCCL from GitHub. 
+We support using a newer version of CCCL with an older version of the CUDA Toolkit, but not the other way around.
+For complete information on compatibility between CCCL and the CUDA Toolkit, see [our platform support](#platform-support).
+
+Everything in CCCL is header-only, so cloning and including it in a simple project is as easy as the following:
+```bash
+git clone https://github.com/NVIDIA/cccl.git
+# Note: You need to use -I and not -isystem in order to ensure you're using the cloned headers and not the ones from the CUDA Toolkit
+nvcc -Icccl/thrust -Icccl/libcudacxx/include -Icccl/cub main.cu -o main 
+```
+
+##### CMake Integration
+
+For more complex projects, we recommend using CMake to integrate CCCL into your project.
+
+CCCL uses [CMake](https://cmake.org/) for all of our build and installation infrastructure, including our tests as well as targets for users to link against their own projects.
+As a result, we recommend anyone using CCCL from GitHub to use CMake to integrate CCCL into your project. 
+For a complete example of how to do this using CMake Package Manager see [our example project](examples/example_project). 
+
+Other build systems should work, but we only test CMake.
+We welcome contributions that would simplify the process of integrating CCCL into other build systems.
+
+### Contributors
+
+Contributor guide coming soon!
+
+## Platform Support
+
+**Objective:** This section describes where users can expect CCCL to compile and run successfully. 
+
+In general, CCCL should work everywhere the CUDA Toolkit is supported, however, the devil is in the details. 
+The sections below describe the details of our support for different versions of the CUDA Toolkit, host compilers, and C++ dialects.
+Furthermore, we describe our testing strategy to be confident that CCCL works everywhere it should. 
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**💡 suggestion:** ‏ 
```suggestion
The sections below describe the details of support and testing for different versions of the CUDA Toolkit, host compilers, and C++ dialects. 
```